docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,698 @@ id: pillars-turbomodules
+
+</TabItem>
+</Tabs>
+
+At the beginning of the spec files are the imports:
+
+- The `TurboModule` type, which defines the base interface for all TurboModules
+- The `TurboModuleRegistry` JavaScript module, which contains functions for loading TurboModules
+
+The second section of the file contains the interface specification for the TurboModule. In this case, the interface defines the `add` function which takes two numbers and returns a promise that resolves to a number. This interface type **must** be named `Spec` for a TurboModule.
+
+Finally, we invoke `TurboModuleRegistry.get`, passing the module's name, which will load the TurboModule if it's available. 
+
+:::caution
+We are writing JavaScript files importing types from libraries, without setting up a proper node module and installing its dependencies. Your IDE will not be able to resolve the import statements and you may see errors and warnings. This is expected and will not cause problems when you add the module to your app.
+:::
+
+## 3. Module Configuration
+
+Next, you need to add some configuration for [**Codegen**](/docs/pillars-codegen.md) and auto-linking.
+
+Some of these configuration files are shared between iOS and Android, while the others are platform-specific.
+
+### Shared
+
+The shared configuration is a `package.json` file that will be used by yarn when installing your module. Create the `package.json` file in the root of the `RTNCalculator` directory.
+
+```json title="package.json"
+{
+  "name": "rtn-calculator",
+  "version": "0.0.1",
+  "description": "Add numbers with TurboModules",
+  "react-native": "js/index",
+  "source": "js/index",
+  "files": [
+    "js",
+    "android",
+    "ios",
+    "rtn-calculator.podspec",
+    "!android/build",
+    "!ios/build",
+    "!**/__tests__",
+    "!**/__fixtures__",
+    "!**/__mocks__"
+  ],
+  "keywords": [
+    "react-native",
+    "ios",
+    "android"
+  ],
+  "repository": "https://github.com/<your_github_handle>/rtn-calculator",
+  "author": "<Your Name> <your_email@your_provider.com> (https://github.com/<your_github_handle>)",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/<your_github_handle>/rtn-calculator/issues"
+  },
+  "homepage": "https://github.com/<your_github_handle>/rtn-calculator#readme",
+  "devDependencies": {},
+  "peerDependencies": {
+    "react": "*",
+    "react-native": "*"
+  },
+  "codegenConfig": {
+    "libraries": [
+      {
+        "name": "RTNCalculatorSpec",
+        "type": "modules",
+        "jsSrcsDir": "js"
+      }
+    ]
+  }
+}
+```
+
+### iOS: Create the `podspec` file
+
+For iOS, you'll need to create a `rtn-calculator.podspec` file which will define the module as a dependency for your app. It will stay in the root of `RTNCalculator`, alongside the `ios` folder.
+
+The file will look like this:
+
+```ruby title="rnt-calculator.podspec"
+require "json"
+
+package = JSON.parse(File.read(File.join(__dir__, "package.json")))
+
+folly_version = '2021.07.22.00'
```

Comment:
React native 69 still comes with `2021.06.28.00-v2`. Not sure whether we want to use the new version: this guide will not work using `2021.07.22`...

Reply:
Yes my bad :( You're actually right. I forgot that this guide is not based upon the current status of `main`. Apologize for the back and forth

Reply:
Addressed